plugins/eclblas/dpotf2.cpp

```diff
@@ -0,0 +1,73 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2016 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+//DPOTF2 computes the Cholesky factorization of a real symmetric
+// positive definite matrix A.
+//The factorization has the form
+// A = U**T * U ,  if UPLO = 'U', or
+// A = L  * L**T,  if UPLO = 'L',
+// where U is an upper triangular matrix and L is lower triangular.
+// This is the unblocked version of the algorithm, calling Level 2 BLAS.
+//
+
+#include "eclblas.hpp"
+#include <math.h>
+
+ECLBLAS_CALL void dpotf2(uint8_t tri, uint32_t r, bool isAllA,
+                         size32_t lenA, const void * A, bool clear,
+                         bool & __isAllResult,
+                         size32_t & __lenResult, void * & __result) {
+  unsigned int cells = r*r;
+  __isAllResult = false;
+  __lenResult = cells * sizeof(double);
+  double *new_a = (double*) rtlMalloc(__lenResult);
+  memcpy(new_a, A, __lenResult);
+  double ajj;
+  // x and y refer to the embedded vectors for the multiply, not an axis
+  unsigned int diag, a_pos, x_pos, y_pos;
+  unsigned int col_step = r;  // between columns
+  unsigned int row_step = 1;  // between rows
+  unsigned int x_step = (tri==UPPER_TRIANGLE)  ? row_step  : col_step;
+  unsigned int y_step = (tri==UPPER_TRIANGLE)  ? col_step  : row_step;
+  for (unsigned int j=0; j<r; j++) {
+    diag = (j * r) + j;    // diagonal
+    x_pos = j * ((tri==UPPER_TRIANGLE) ? col_step  : row_step);
```

Comment:
I think it would be less clear.  The x_step and y_step variables are the number of positions to get to the next element used in the dot product function.  The next y value depends upon whether you are going down a column or across a row.  The y_step happens to be useful to calculate the first y_pos for the dot product.  The x_step is only useful to calculate the next x_pos in the dot product function.

So, I would like to keep them clearly separate for the reader.